components/molecule/table/src/index.js

```diff
@@ -0,0 +1,127 @@
+import React, {lazy, Suspense} from 'react'
+import PropTypes from 'prop-types'
+
+const Table = lazy(() => import('antd/lib/table'))
+
+const BASE_CLASS = `sui-MoleculeTable`
+const CLASS_ACTIONS = `${BASE_CLASS}-actions`
+const CLASS_MOBILE_MODE = `${BASE_CLASS}-mobile`
+const CLASS_MOBILE_MODE_ROW = `${CLASS_MOBILE_MODE}-row`
+const CLASS_MOBILE_MODE_CELL = `${CLASS_MOBILE_MODE_ROW}-cell`
+const CLASS_MOBILE_MODE_CELL_TITLE = `${CLASS_MOBILE_MODE_CELL}-title`
+const CLASS_MOBILE_MODE_CELL_CONTENT = `${CLASS_MOBILE_MODE_CELL}-content`
+
+const getActionsHeaderConfig = actions => ({
+  title: 'Actions',
+  key: 'actions',
+  align: 'left',
+  width: 150,
+  render: (text, record) => {
+    const extendedActions = React.cloneElement(actions, {
+      text,
+      record
+    })
+
+    return <span className={CLASS_ACTIONS}>{extendedActions}</span>
+  }
+})
+
+const renderMobileMode = ({title, dataSource, columns}) => { // eslint-disable-line
+  return (
+    <div className={CLASS_MOBILE_MODE}>
+      <h2>{title}</h2>
+      {dataSource.map((row, indexRow) => {
+        return (
+          <div key={indexRow} className={CLASS_MOBILE_MODE_ROW}>
+            {Object.keys(row).map((keyTitle, index) => {
+              const column = columns.filter(
+                column => column.key === keyTitle
+              )[0]
+
+              let dataCell = ''
+              if (typeof row[keyTitle] === 'object' && column.render) {
+                dataCell = column.render(row[keyTitle])
+              } else {
+                dataCell = row[keyTitle]
+              }
+
+              return (
+                <div key={index} className={CLASS_MOBILE_MODE_CELL}>
+                  <div className={CLASS_MOBILE_MODE_CELL_TITLE}>
+                    {column.title}
+                  </div>
+                  <div className={CLASS_MOBILE_MODE_CELL_CONTENT}>
+                    {dataCell}
+                  </div>
+                </div>
+              )
+            })}
+          </div>
+        )
+      })}
+    </div>
+  )
+}
+
+const MoleculeTable = ({
+  dataSource,
+  actions,
+  title,
+  mobile,
+  fixedHeader,
+  columns: originalColumns,
+  ...props
+}) => {
+  const columns = actions
+    ? [...originalColumns, getActionsHeaderConfig(actions)]
+    : originalColumns
+
+  return (
+    <div className={BASE_CLASS}>
+      {mobile ? (
+        renderMobileMode({title, dataSource, columns})
```

Comment:
Why mobile mode depends on a prop? Shouldn't be the component itself "auto" responsive?

Reply:
Not really. Because of some situations when doing SSR it was agreed w/ the enablers that a better strategy is to delegate to `LayoutMediaQuery` all the responsibility of detecting the size of the viewport so we can display the right version of any component via props
Maybe @jcger @carlosvillu or @midudev can give you more info about this